### DISCUSSION TITLE
ERR_EMPTY_RESPONSE from localhost:8091

### DISCUSSION THREAD

**Discussion by bikemike** [zwave-js/zwave-js-ui#3425] [General]
I've installed v9.3.2 through snap and the web UI was working initially. Unfortunately it has stopped working and I'm not sure why. Chrome gives me the following:

http://localhost:8091
```
This page isn’t working
localhost didn’t send any data.
ERR_EMPTY_RESPONSE
```

I had also tried to set up https in the settings prior to this but it doesn't work either. I assume it's also an empty response:

https://localhost:8091
```
Your connection is not private
Attackers might be trying to steal your information from localhost (for example, passwords, messages, or credit cards). [Learn more](chrome-error://chromewebdata/#)
NET::ERR_CERT_INVALID
```

Is there a place I can look for connection and error logging to see what might be causing this issue?

Thanks!

**Comment by robertsLando** [comment]
Could you attach application logs please?

**Reply by bikemike** [reply]
I would love to but I need some help figuring out how to do that.

The documentation says logging is enabled though the UI which, in my case, is not functioning.

**Reply by robertsLando** [reply]
@bikemike You should go to the store directory, `logs` folder and take zui logs files. I dunno on snap which is the default store folder, @jmgiaever could help with that as he is the snap package maintainer

**Reply by jmgiaever** [reply]
Typically `/var/snap/zwave-js-ui/<current revision>` where `current revision` is the revision in use. There should be a symlink named `current` which links to this revision number. 

So you can try `/var/snap/zwave-js-ui/current`. If not you have to find it, e.g with `snap info zwave-js-ui | grep installed` and locate the number within the parentheses.

**Reply by bikemike** [reply]
[zwavejs_2023-11-15.log](https://github.com/zwave-js/zwave-js-ui/files/13367626/zwavejs_2023-11-15.log)

I've attached the log file. I don't think it contains any helpful information but hopefully I'm wrong.

**Reply by jmgiaever** [reply]
Could you give a copy paste of the command:

```
sudo snap get -d zwave-js-ui
```

too?

**Reply by bikemike** [reply]
Here it is:
```
{
	"mqtt": {
		"name": ""
	},
	"server": {
		"force-disable-ssl": false,
		"host": "0.0.0.0",
		"port": 8091,
		"ssl": false
	},
	"session": {
		"cookie-secure": "",
		"secret": "2faef3ce-7abd-11ee-a330-5b9c7013b28c"
	},
	"timezone": "America/Vancouver"
}

```

**Reply by jmgiaever** [reply]
Looks quite the same as mine, except I have `force-disable-ssl` ser to `true`. But not sure if that would do anything.

You can try with
```
sudo snap set zwave-js-ui server.force-disable-ssl=true
```

**Reply by bikemike** [reply]
That did the trick. Thank you!!

I'm pretty sure I had set an SSL or HTTPS option in the UI but I can't find it now (removed in the latest version?). Maybe it was removed because of this very reason?

**Reply by jmgiaever** [reply]
@robertsLando

**Reply by kpine** [reply]
@jmgiaever Looks like you are [setting](https://github.com/giaever-online-iot/zwave-js-ui/blob/5bcee67dca3e20d004781fd8cab63bb07f3d3eb5/src/helper/env-wrapper#L109C8-L109C18) `SERVER_SSL`, which is not a true/false value. If the env var is set to anything, it forces SSL to be enabled. This environment variable is only documented for use by developers. https://zwave-js.github.io/zwave-js-ui/#/development/intro?id=developing-against-a-different-backend

**Reply by robertsLando** [reply]
`SERVER_SSL` is pretty useless, it only makes sense when running `vite` in dev mode with hot reloading.

**Reply by robertsLando** [reply]
> I'm pretty sure I had set an SSL or HTTPS option in the UI but I can't find it now (removed in the latest version?). Maybe it was removed because of this very reason?

@bikemike It's still there under `General` section in settings

**Reply by kpine** [reply]
> `SERVER_SSL` is pretty useless, it only makes sense when running `vite` in dev mode with hot reloading.

Same for `SERVER_HOST` and `SERVER_PORT`?

**Reply by bikemike** [reply]
@robertsLando, this is my Settings->General section:

<img width="700" alt="image" src="https://github.com/zwave-js/zwave-js-ui/assets/7464404/cb4b250c-c265-4deb-855b-ab966fb3a486">

**Reply by bikemike** [reply]
@robertsLando, I hit the refresh button then quickly clicked general and saw the option for a brief moment before it disappeared.

<img width="788" alt="image" src="https://github.com/zwave-js/zwave-js-ui/assets/7464404/9b7017ef-559d-4a7d-8a9b-0122244ed342">

**Reply by kpine** [reply]
If you enable `FORCE_DISABLE_SSL` then the option is hidden from the UI.

**Reply by robertsLando** [reply]
> If you enable FORCE_DISABLE_SSL then the option is hidden from the UI.

Exactly. The reason why you see it on startup for a short time is that it is hidden once I receive the settings from the api call

**Reply by bikemike** [reply]
Ahh. And that's what @jmgiaever had me do. I see. Thanks! Still wondering how setting the FORCE_DISABLE_SSL fixed my ERR_EMPTY_RESPONSE issue but happy to have the UI back.

**Reply by kpine** [reply]
Because you were accessing an SSL endpoint using non-SSL, and setting that to `true` disables SSL. Your configuration forced SSL **on** due to the snap usage of env var `SERVER_SSL`, despite the settings.

If you can edit the snap configuration, **deleting** fields `"force-disable-ssl"` and `"ssl"`, it should work w/o SSL.

```console
❯ curl http://github.com:443
curl: (52) Empty reply from server
```

**Reply by bikemike** [reply]
Where and how is SERVER_SSL set? I don't believe I ever set it. I think I might have selected the HTTPS option in the general settings but querying them in the terminal (sudo snap get -d zwave-js-ui) showed "ssl": false. If it was enabled, wouldn't https://localhost:8091 have worked?

**Reply by kpine** [reply]
You're not setting it, the snap is. It's a problem with the snap. https://github.com/zwave-js/zwave-js-ui/discussions/3425#discussioncomment-7582880

According to your comment, SSL was working:

> Your connection is not private
> Attackers might be trying to steal your information from localhost (for example, passwords, messages, or credit cards). [Learn more](chrome-error://chromewebdata/#)
> NET::ERR_CERT_INVALID

Here your problem is Chrome. It doesn't allow localhost connections using self-signed certificates. It works fine in Firefox once you tell it to accept the certificate. You can either create a valid certificate and install it into ZUI, or force Chrome to allow "unsafe" localhost connections: https://communicode.io/allow-https-localhost-chrome/
